exp-player/addon/mixins/warn-on-exit-controller.js

```diff
@@ -0,0 +1,33 @@
+import Ember from 'ember';
+
+export default Ember.Mixin.create(Ember.Evented, {
+    showWarning: false,
+    forceExit: false,
+
+    session: Ember.computed(function() {
+        return this.get('model.session') || this.get('model');
```

Comment:
This will get simplified by https://github.com/CenterForOpenScience/exp-addons/pull/51, I.E. `session` will always be set on the controller.

The existing approach has the fallback mostly to cooperate with the lookit implementation